bindings/experimental/distrdf/python/DistRDF/Ranges.py

```diff
@@ -29,7 +29,7 @@ class EmptySourceRange(object):
 
     def __init__(self, start, end):
         """set attributes"""
-        self.id = self._ids.next()
+        self.id = self._ids.__next__()
```

Comment:
I think idiomatic Python would be `next(self._ids)`?

Reply:
I thought that too, that's what I did in the beginning. Then deepcode warned against it and told me to use `__next__` . Unfortunately I cannot show you the failing deepcode build since apparently it doesn't keep a history of it but overwrites the same comment ( The deepcode comment you see above that is now "green" it used to be "red").

Reply:
But since we all agree on `next` I will revert this commit :+1:

Reply:
Weird that deepcode complains about it? Let's see what it says exactly.

Reply:
ah yeah i just saw the rest of the commits and was just going to ask why you changed it :smile:

Reply:
btw I _think_ `iterator.__next__` is py3 only?

Reply:
:scream: good catch! That's more than enough to ditch whatever deepcode thinks (provided it does still think what it thought 2 days ago)

Reply:
Ok so deepcode is still thinking I should guard the call to `next`, but it gives `iterator.next` as a possible fix in its dashboard https://www.deepcode.ai/app/gh/root-project/root/aa21d63ca3a1f4cdb2b559e4cb1c2c2d7eb65f34/root-project/root/5c5eb503b4de98306588af031279c88c18375162/pr/_/%2Fbindings%2Fexperimental%2Fdistrdf%2Fpython%2FDistRDF%2FRanges.py/python%2Fdc_interfile_project%2Funguarded~next~call/code/?utm_source=gh_review

Now I also remember why I wrote `iterator.next` in the first commit after this warning. So this "solution" came from some Pyhton2 repository